@here/harp-map-controls/test/MapControlsTest.ts

```diff
@@ -530,4 +538,93 @@ describe("MapControls", function () {
+                // `await` on because it is called in the constructor, but we can disable them being
+                // added.
+                addBackgroundDatasource: false,
+                enablePolarDataSource: false
+            };
+            mapView = new MapView(mapViewOptions);
+            mapControls = new MapControls(mapView);
+        });
+
+        it("toggle tilt reaches configured tilt angle and 0", async () => {
+            mapControls.inertiaEnabled = true;
+            mapControls.toggleTilt();
+
+            let resolvePromise: (value?: unknown) => void;
+            let rejectPromise: (value?: unknown) => void;
+
+            {
+                let previousTilt: number | undefined;
+
+                // Tilt to "mapControls.tiltAngle"
+                const tiltCamera = new Promise((resolve, reject) => {
+                    resolvePromise = resolve;
+                    rejectPromise = reject;
+                });
+                const checkReachesTiltAngle = () => {
+                    const mapViewTiltRad = THREE.MathUtils.degToRad(mapView.tilt);
+
+                    // Resolve when we have reached within EPSILON of the tiltAngle
+                    if (Math.abs(mapViewTiltRad - mapControls.tiltAngle) <= Number.EPSILON) {
+                        mapView.removeEventListener(
+                            MapViewEventNames.AfterRender,
+                            checkReachesTiltAngle
+                        );
+                        resolvePromise();
+                    } else if (previousTilt === mapViewTiltRad) {
+                        rejectPromise();
+                    }
+                    previousTilt = mapViewTiltRad;
+                };
+                mapView.addEventListener(MapViewEventNames.AfterRender, checkReachesTiltAngle);
+                await tiltCamera;
+            }
+            {
+                let previousTilt: number | undefined;
+
+                // Go back to zero.
+                const tiltBackToZero = new Promise((resolve, reject) => {
+                    resolvePromise = resolve;
+                    rejectPromise = reject;
+                });
+                mapControls.toggleTilt();
+                const checkGoesToZero = () => {
+                    const mapViewTiltRad = THREE.MathUtils.degToRad(mapView.tilt);
+
+                    // Resolve when we reach 0
+                    if (mapViewTiltRad <= Number.EPSILON) {
+                        mapView.removeEventListener(MapViewEventNames.AfterRender, checkGoesToZero);
+                        resolvePromise();
+                    } else if (previousTilt === mapViewTiltRad) {
+                        rejectPromise();
+                    }
+                    previousTilt = mapViewTiltRad;
+                };
+                mapView.addEventListener(MapViewEventNames.AfterRender, checkGoesToZero);
+                await tiltBackToZero;
```

Comment:
```suggestion
                await tiltBackToZero;
```
return expect(tiltBackToZero).to.eventually.be.closeTo(0, Number.EPSILON);

Reply:
I have changed to use expect.

I don't like the return syntax, because otherwise the test gets more nested and it makes readability / maintenance harder IMO, also, await is used a lot in existing tests, so I think this is not some exception.